src/services/fileStorage/docs/index.js

```diff
@@ -1,200 +1,895 @@
-				},
-			],
-			summary: 'Retrieve all files and immediate sub-directories for the given path',
-		},
-		remove: {
-			parameters: [
-				{
-					description: 'The path where the file can be found',
-					required: true,
-					name: 'path',
-					type: 'string',
-				},
-			],
-			summary: 'remove a file',
-		},
-		patch: {
-			parameters: [
-				{
-					description: 'The name of the file',
-					required: true,
-					name: 'fileName',
-					type: 'string',
-				},
-				{
-					description: 'The path where the file can be found, have to have a slash at the end',
-					required: true,
-					name: 'path',
-					type: 'string',
-				},
-				{
-					description: 'The path where the file should be moved to, have to have a slash at the end',
-					required: true,
-					name: 'destination',
-					type: 'string',
-				},
-				{
-					description: 'The id of the file in the proxy db',
-					required: true,
-					name: 'id',
-					type: 'string',
-				},
-			],
-			summary: 'moves a file to a new path',
+		description: '',
+		operations: {
+			create: {
+				parameters: [
+					{
+						description: 'Filename',
+						name: 'name',
+						type: 'String',
+						required: true,
+					},
+					{
+						description: 'ID of either an course or team. For user files leave out.',
+						type: 'String (ObjectID)',
+						name: 'owner',
+					},
+					{
+						description: 'ID of parent folder',
+						type: 'String (ObjectID)',
+						name: 'parent',
+					},
+					{
+						description: 'Mimetype of file',
+						type: 'String',
+						name: 'type',
+					},
+					{
+						description: 'Filesize in Byte',
+						name: 'size',
+						type: 'Number',
+						schema: {
+							type: 'integer',
+							format: 'int32',
+							minimum: 0,
+						},
+					},
+					{
+						description: 'Filename aka identifier in storage context',
+						name: 'storageFileName',
+						type: 'String',
+						required: true,
+						schema: {
+							type: 'string',
+						},
+					},
+					{
+						description: 'Thumbnail of file',
+						type: 'String',
+						name: 'thumbnail',
+					},
+					{
+						description: 'Determine if students can edit if file is a course file',
+						type: 'Boolean',
+						name: 'studentCanEdit',
+						schema: {
+							type: 'boolean',
+						},
+					},
+					{
+						description: 'Override default permissions on creation',
+						name: 'permissions',
+						type: 'Array',
+						style: 'matrix',
+						explode: true,
+						schema: {
+							type: 'array',
+							items: {
+								type: 'object',
+							},
+						},
+					},
+				],
+				description: '',
+				summary: 'Creates a new file object',
+				responses: {
+					200: {
+						description: 'Returns the data of the newly created file',
+						example: {
+							name: '24-1.gif',
+							owner: '0000dcfbfb5c7a3f00bf21ab',
+							type: 'image/gif',
+							size: 2545,
+							storageFileName: '1560977688023-24-1.gif',
+							thumbnail: 'https://schulcloud.org/images/login-right.png',
```

Comment:
shema should not be part of this, start with //

Reply:
This is just an example.